### PR TITLE
Port the crate to the `piecrust`-enabled ABI

### LINKED ISSUE #69: Port to be compatible with `piecrust` contract
**Describe what you want implemented**
To ensure that a transaction can be passed correctly through the `piecrust`-enabled transfer contract, the `Transaction` structure introduced  in the to-merge PR - https://github.com/dusk-network/phoenix-core/pull/117 - should be used, and the one provided here removed.

**Describe "Why" this is needed**
This is necessary to ensure that the phoenix transaction model is properly used, and the transaction can pass through the boundary of the VM. 

**Describe alternatives you've considered**
N/A

**Additional context**
This will likely make the `canonical` dependency obsolete.

### LINKED ISSUE #70: Remove `TransactionSkeleton`
**Describe what you want implemented**
Since we're implementing the transaction structure elsewhere - namely [`phoenix-core`](https://github.com/dusk-network/phoenix-core/pull/117) - and implementing hashing functionality there, the `TransactionSkeleton` structure may be removed.

**Describe "Why" this is needed**
It is always good to remove unused code.

**Describe alternatives you've considered**
N/A

**Additional context**
This is part of an effort to standardize how hashing is done across our stack.

### PR DESCRIPTION
**Pull request by ureeves**
Resolves #69
Resolves #70

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,24 +9,22 @@ license = "MPL-2.0"
...
 
 [dev-dependencies]
 rand = "^0.8"
-canonical_derive = "0.7"
 
 [lib]
 crate-type = ["cdylib", "rlib"]
+
+[patch.crates-io]
+rusk-abi = { path = "../rusk/rusk-abi" }
```
Comment (moCello) [inline Cargo.toml:30]: I assume this patch is needed until `ruk-abi` is released?

**Review by moCello — APPROVED** [review]
LGTM but I propose someone more familiar with the crate to also have a look

**Review by herr-seppia — CHANGES_REQUESTED** [review]
LGTM Overall

- A question to be addressed (the naming one) before approve
- The rest is optional

File: Cargo.toml (hunk 1 above)
Comment (ureeves) [reply Cargo.toml]: Removed the patch in favor of `rusk-abi` `0.8.0-alpha`

**Review by herr-seppia — APPROVED** [review]
Ad discussed I am going to approve under the condition that the nomenclature comment is addressed

**Comment by ureeves** [comment]
This is blocked by the release of a `rusk-abi` version compatible with https://github.com/dusk-network/piecrust/pull/223

File: Cargo.toml [hunk 2]
```diff
@@ -9,24 +9,20 @@ license = "MPL-2.0"
...
-canonical = "0.7"
-dusk-bls12_381-sign = { version = "0.3.0-rc", default-features = false, features = [
-  "canon",
-] }
+dusk-schnorr = { version = "0.12", default-features = false }
+dusk-jubjub = { version = "0.12", default-features = false }
+dusk-poseidon = { version = "0.29.1-rc.0", default-features = false }
+dusk-merkle = { version = "0.4.1-rc.0", features = ["poseidon", "rkyv-impl"] }
+dusk-plonk = { version = "0.13", default-features = false }
+rusk-abi = { version = "0.8.0-piecrust.0.5", features = ["host"] }
```
Comment (herr-seppia) [inline Cargo.toml:20]: @ureeves is this still required?

cc @ZER0

Comment (ureeves) [reply Cargo.toml:20]: The "host" feature will cease to be required when I complete the port of `rusk-abi` to `piecrust@0.6`

Comment (herr-seppia) [reply Cargo.toml:20]: Perfect!

**Comment by ureeves** [comment]
Irrelevant since #77 and #78 were merged

**Pull request closed without merging**